### ISSUE TITLE
[Bug] Improve MSAL logging around ESTS-R failures

### ISSUE THREAD

**Issue by jennyf19** [AzureAD/microsoft-authentication-library-for-dotnet#3274]
Version: 4.36.1

Confidential client initialization can fail intermittently in some case, and fallback to global not working out. MSAL error and warning logs are not sufficient to root cause. 

TODOs:
1. Investigate why the fallback to global may not work in some cases and can the customer issue be reproed?
2. Improve logging in affected area, so log messages provide more details to root cause

Example error message:
```MSAL.Desktop.4.36.1.0.MsalServiceException: 
ErrorCode: invalid_instance
Microsoft.Identity.Client.MsalServiceException: AADSTS50049: Unknown or invalid instance.
```

Example corresponding logs:
```{"RetryCount":"0","{OriginalFormat}":"[ConfidentialClientApplication.cs:LoggingCallback:100] False MSAL 4.36.1.0 MSAL.Desktop 4.7.2 Windows Server 2019 (ConfidentialClientApplication:GUID:https://login.windows.net/GUID: 0.0.0.0) [Region discovery] Auto-discovery failed."}
{"RetryCount":"0","{OriginalFormat}":"[ConfidentialClientApplication.cs:LoggingCallback:100] False MSAL 4.36.1.0 MSAL.Desktop 4.7.2 Windows Server 2019 (ConfidentialClientApplication:GUID:https://login.windows.net/GUID: 0.0.0.0) [Region discovery] Auto-discovery failed."}
{"RetryCount":"0","{OriginalFormat}":"[ConfidentialClientApplication.cs:LoggingCallback:100] False MSAL 4.36.1.0 MSAL.Desktop 4.7.2 Windows Server 2019 (ConfidentialClientApplication:GUID:https://login.windows.net/GUID: 0.0.0.0) Request retry failed."}
{"RetryCount":"0","{OriginalFormat}":"[ConfidentialClientApplication.cs:LoggingCallback:103] False MSAL 4.36.1.0 MSAL.Desktop 4.7.2 Windows Server 2019 (ConfidentialClientApplication:GUID:https://login.windows.net/GUID: 0.0.0.0) Exception type: Microsoft.Identity.Client.MsalServiceException\r\n, ErrorCode: invalid_instance\r\nHTTP StatusCode 400\r\nCorrelationId GUID\r\n"}
{"RetryCount":"0","{OriginalFormat}":"[ConfidentialClientApplication.cs:LoggingCallback:103] False MSAL 4.36.1.0 MSAL.Desktop 4.7.2 Windows Server 2019 (ConfidentialClientApplication:GUID:https://login.windows.net/GUID: 0.0.0.0) [Instance Discovery] Instance discovery failed - invalid instance!"}
```

**Comment by bgavrilMS** [comment]
This has already been addressed by #2975 and #3014

**Comment by bgavrilMS** [comment]
If regional fails, this should log warnings instead of info. Will follow up as part of https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/4388
